modopt/opt/algorithms.py

```diff
@@ -502,8 +519,8 @@ def __init__(self, x, grad, prox, cost='auto', beta_param=1.0,
 
         # Set the initial variable values
         self._check_input_data(x)
-        self._x_old = np.copy(x)
-        self._z_old = np.copy(x)
+        self._x_old = self.copy_data(x)
+        self._z_old = self.copy_data(x)
```

Comment:
As this issue focusses only on ForwardBackward, I have added move data to GPU function only here.
Again, if use_gpu is False, it is still np.copy which is done.